R/process_extract_gp_ooh.R

```diff
@@ -82,6 +82,10 @@ process_extract_gp_ooh <- function(year, data_list, write_to_disk = TRUE) {
     dplyr::mutate(
       record_keydate1 = as.Date(.data$record_keydate1),
       record_keydate2 = as.Date(.data$record_keydate2)
+    ) %>%
+    # change attendance_status into numeric
+    mutate(
+      attendance_status = as.double(attendance_status)
```

Comment:
I don't think this will work properly as it's already a factor from [`read_extract_ooh...`](https://github.com/Public-Health-Scotland/source-linkage-files/blob/64c51db19a051ce173620af372942a7b17b926e4/R/read_extract_ooh_consultations.R#L16) so when you convert it (back?) to a numeric, it will become `1` or `2` but actually we want it to match the `attendance_status` [values for outpatients](https://www.ndc.scot.nhs.uk/Dictionary-A-Z/Definitions/index.asp?Search=A&ID=96&Title=Attendance%20Status) which are 1 (seen) or 8 (not seen).

Reply:
We should probably change the GP OoH `attendance_status` (`"Consultation Recorded"`) and the outpatient `attendance_status` ("Clinic Attendance (Status) Code") to both be a factor with something like:
`factor(<ooh_var>, levels = c("Y", "N"), labels = c("Patient Seen", "DNA"))` and
`factor(<outpatient_var>, levels = c(1, 5, 8), labels = c("Patient Seen", "CNW", "DNA"))`.

But this would also involve tweaking some other code which uses the 1 / 8 values, so out of scope for this PR. For now, I think it's sufficient to alter the GP OoH values this just needs to be done at the right place.